Sources/Controllers/TravelGuides/explore/SavedArticlesTabViewController.swift

```diff
@@ -8,72 +8,277 @@
             }
-            outCell = cell
         }
-        return outCell
+        return UITableViewCell()
     }
     
-    override func onRowSelected(_ indexPath: IndexPath!) {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         let item = tableData.item(for: indexPath)
+        lastSelectedIndexPath = indexPath
         if let article = item.obj(forKey: "article") as? TravelArticle {
-            if let tabViewDelegate {
-                tabViewDelegate.openArticle(article: article, lang: article.lang)
+            OAAppSettings.sharedManager().travelGuidesState.article = article
+            OAAppSettings.sharedManager().travelGuidesState.wasOpenedFromBookmarks = true
+            let vc = TravelArticleDialogViewController(articleId: article.generateIdentifier(), lang: article.lang ?? "")
+            vc.delegate = self
+            navigationController?.pushViewController(vc, animated: true)
+        }
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
+    func tableView(_ tableView: UITableView, contextMenuConfigurationForRowAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        let item = tableData.item(for: indexPath)
+        if item.cellType == ArticleTravelCell.getIdentifier() {
+            if let article = item.obj(forKey: "article") as? TravelArticle {
+                let lang = item.string(forKey: "lang") ?? ""
+                
+                let menuProvider: UIContextMenuActionProvider = { _ in
+                    let readAction = UIAction(title: localizedString("shared_string_read"), image: UIImage(systemName: "newspaper")) { [weak self] _ in
+                        guard let self else { return }
+                        lastSelectedIndexPath = indexPath
+                        OAAppSettings.sharedManager().travelGuidesState.article = article
+                        OAAppSettings.sharedManager().travelGuidesState.wasOpenedFromBookmarks = true
+                        let vc = TravelArticleDialogViewController(articleId: article.generateIdentifier(), lang: article.lang ?? "")
+                        vc.delegate = self
+                        navigationController?.pushViewController(vc, animated: true)
+                    }
+                    let bookmarkAction = UIAction(title: localizedString("shared_string_remove_bookmark"), image: UIImage(systemName: "bookmark")) { [weak self] _ in
+                        guard let self else { return }
+                        TravelObfHelper.shared.getBookmarksHelper().removeArticleFromSaved(article: article)
+                        self.generateData()
+                        self.tableView.deleteRows(at: [indexPath], with: .automatic)
+                    }
+                    let pointsAction = UIAction(title: localizedString("shared_string_gpx_points"), image: UIImage(named: "point.topleft.filled.down.to.point.bottomright.curvepath")) { [weak self] _ in
```

Comment:
point.topleft.filled.down.to.point.bottomright.curvepath - It's a very long name.:)